src/C++/Session.cpp

```diff
@@ -915,7 +936,10 @@ void Session::generateReject( const Message& message, const std::string& str )
 
   message.getHeader().getField( msgType );
   message.getHeader().getField( msgSeqNum );
-  if ( beginString >= FIX::BeginString_FIX42 )
+
+  const DataDictionary& sessionDD = m_dataDictionaryProvider.getSessionDataDictionary(m_sessionID.getBeginString());
+
+  if ( sessionDD.isMsgField(MsgType("3"), FIELD::RefMsgType) )
```

Comment:
Changes check to support custom FIX dictionaries as well that may have added the field RefMsgType to their reject messages